### PR TITLE
Fix ubuntu builds

### PR DESCRIPTION
**Pull request by NeilW**
Fix ubuntu builds by binding in the host resolver when using the host network namespace

Update the default distribution to a supported version.

### CODE REVIEW

File: mkosi [hunk 1]
```diff
@@ -741,7 +741,10 @@ def run_workspace_command(args, workspace, *cmd, network=False, env={}):
                "--register=no",
                "--bind=" + var_tmp(workspace) + ":/var/tmp" ]
 
-    if not network:
+    if network:
+        # If we're using the host network namespace, use the same resolver
+        cmdline += ["--bind-ro=/etc/resolv.conf"]
```
Comment (lucaswerkmeister) [inline mkosi]: Hm, what does this do if `/etc/resolv.conf` is a symlink?

Comment (NeilW) [reply mkosi]: Well here the links are followed on both sides and mounted in the appropriate place. 

/dev/vda1 on /run/systemd/resolve/stub-resolv.conf type ext4 (ro,relatime,seclab
el,data=ordered)

Comment (lucaswerkmeister) [reply mkosi]: Hm, looks like it indeed work.

Comment (juaningan) [reply mkosi]: I think that same change is missing also for build scipt

**Pull request merged**